proto/proto/stake.proto

```diff
@@ -69,6 +76,12 @@ message Delegate {
   // The delegation amount, in units of unbonded stake.
   // TODO: use flow aggregation to hide this, replacing it with bytes amount_ciphertext;
   uint64 unbonded_amount = 3;
+  // The amount of delegation tokens produced by this action.
+  // 
+  // This is implied by the validator's exchange rate in the specified epoch
+  // (and should be checked in transaction validation!), but including it allows
+  // stateless verification that the transaction is internally consistent.
+  uint64 delegation_amount = 4; 
```

Comment:
To compute the action's contributions to the value balance, we need to compute the delegation amount, and this is only possible with knowledge of all of the validator exchange rates.  In principle, we could omit this data to save a few bytes, but doing so means that we wouldn't be able to statelessly verify the rest of the transaction data.  Instead, by adding it, we add a required consistency check to stateful verification, but allow stateless verification of the transaction data.